src/black/__init__.py

```diff
@@ -270,14 +280,21 @@ def get_cache_key(self) -> str:
             version_str,
             str(self.line_length),
             str(int(self.string_normalization)),
-            str(int(self.is_pyi)),
+            str(int(self.source_type)),
         ]
         return ".".join(parts)
 
 
 # Legacy name, left for integrations.
 FileMode = Mode
 
+def lazy_import_json() -> None:
+    global json
+    if json is None:
+        try:
+            import ujson as json
```

Comment:
Maybe add `ujson` as an extra install then and document somewhere that it can be used to speed up ipython usage.

Reply:
Added an extra `notebook_speedup`